src/shared/libs/rest/middleware/parse-token.middleware.ts

```diff
@@ -0,0 +1,49 @@
+import { NextFunction, Request, Response } from 'express';
+import { jwtVerify } from 'jose';
+import { StatusCodes } from 'http-status-codes';
+
+import { createSecretKey } from 'node:crypto';
+
+import { Middleware } from './middleware.interface.js';
+import { HttpError } from '../errors/http-error.js';
+import { TokenPayload } from '../../../modules/auth/index.js';
+
+function isTokenPayload(payload: unknown): payload is TokenPayload {
+  return (
+    (typeof payload === 'object' && payload !== null) &&
+    ('email' in payload && typeof payload.email === 'string') &&
+    ('name' in payload && typeof payload.name === 'string') &&
+    ('id' in payload && typeof payload.id === 'string')
+  );
```

Comment:
Можно сократить проверку.
Если атрибута нет, то `typeof` вернет `undefined`
```suggestion
  return (
    (typeof payload === 'object' && payload !== null) &&
    typeof payload?.email === 'string' &&
    typeof payload?.name === 'string' &&
    typeof payload?.id === 'string'
  );
```